readme.MD

```diff
@@ -1,16 +1,47 @@
 # Dialogs Framework
 The Dialogs Framework is a Python library for writing resumable functions with persistent execution state.
 
-# Examples
+## Resumable what now?
+This framework is intended for long-lived backend flows, such as chatbot conversations. It allows one to write code as a chronological flow, and provides the magic to make execution resume where it left off.
 
-## Set up
+## Code sample
+```python
+# No dialog, manual state manipulation
+def ask_a_name(client_answer, state):
+    if not state.did_ask_already:
+        state.did_ask_already = True
+        name = client_answer
+        return f"{name} is a beautiful name!"
+    else:
+        return "What is your name?"
+
+# With the dialogs framework
+@dialog()
+def ask_a_name_dialog():
+    run(send_message("What is your name?"))
+    name = run(get_client_response())
+    run(send_message(f"{name} is a beautiful name!"))
+```
+
+## Philosophy
+* No direct access to flow state.
+* No breaks in the flow with `return` statements.
+* No in-memory state, scale-out friendly.
```

Comment:
You should explain what flow is

Reply:
Removed it from the first point, not sure how to simplify the second